ansible/provision-pulsar-aws.tf

```diff
@@ -0,0 +1,117 @@
+variable "public_key_path" {
+  description = <<DESCRIPTION
+Path to the SSH public key to be used for authentication.
+Ensure this keypair is added to your local SSH agent so provisioners can
+connect.
+
+Example: ~/.ssh/terraform.pub
+DESCRIPTION
+}
+
+variable "key_name" {
+  description = "Desired name of AWS key pair"
+}
+
+variable "region" {
+    default = "us-west-2"
+}
+
+variable "ami" {
+    default = "ami-9fa343e7" // RHEL-7.4
+}
+
+provider "aws" {
+    region     = "${var.region}"
+}
+
+# Create a VPC to launch our instances into
+resource "aws_vpc" "pulsar_vpc" {
+  cidr_block = "10.0.0.0/16"
+
+  tags {
+      Name = "Pulsar-VPC"
+  }
+}
+
+# Create an internet gateway to give our subnet access to the outside world
+resource "aws_internet_gateway" "default" {
+  vpc_id = "${aws_vpc.pulsar_vpc.id}"
+}
+
+# Grant the VPC internet access on its main route table
+resource "aws_route" "internet_access" {
+  route_table_id         = "${aws_vpc.pulsar_vpc.main_route_table_id}"
+  destination_cidr_block = "0.0.0.0/0"
+  gateway_id             = "${aws_internet_gateway.default.id}"
+}
+
+# Create a subnet to launch our instances into
+resource "aws_subnet" "pulsar_subnet" {
+  vpc_id                  = "${aws_vpc.pulsar_vpc.id}"
+  cidr_block              = "10.0.0.0/24"
+  map_public_ip_on_launch = true
+}
+
+resource "aws_security_group" "pulsar_security_group" {
+  name        = "terraform"
+  vpc_id      = "${aws_vpc.pulsar_vpc.id}"
+
+  # SSH access from anywhere
+  ingress {
+    from_port   = 22
+    to_port     = 22
+    protocol    = "tcp"
+    cidr_blocks = ["0.0.0.0/0"]
+  }
+
+  # All ports open within the VPC
+  ingress {
```

Comment:
Best practice is to open only the required ports; even within the VPC

Reply:
@afalko To be clear, this PR is a work in progress. I'll ping you when it's ready for review.

Reply:
Thanks! Sorry if I jumped the gun; I had noted similar feedback in the messaging-benchmark and was happy to see a PR where I could communicate it :).

Reply:
@afalko No worries! I'm still working on getting the playbook to run. Once I get over that hurdle we can start digging in :)